### PR TITLE
Backend app/feature/oauth

### PR DESCRIPTION
**Pull request by kiselev-danil**
Added routings with oauth authentication for token verification tests. Small fixes for token verification and oauth configuration. User's routings moved into their own file

### CODE REVIEW

**Review by 1MPULSEONE — APPROVED** [review]

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/TokenVerifier.kt [hunk 1]
```diff
@@ -4,29 +4,36 @@ import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken
...
         if ((acceptableMailDomain == hostedDomain) && emailVerified) {
             userMail = payload.email
         }
         return userMail ?: throw Exception("Token wasn't verified")
     }
 
+    private fun extractDomain(email: String?): String {
+        email ?: throw Exception("Email is empty")
+        return email.split('@').last()
+
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/TokenVerifier.kt:36]: Usless space

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt [hunk 2]
```diff
@@ -12,27 +13,31 @@ import office.effective.features.user.service.IUserService
...
-        val tokenStr = call.request.header("id_token") ?: call.response.status(HttpStatusCode.Forbidden)
-        val user = service.getUserById(userId as String, tokenStr as String)
-        call.respond(user ?: "No such user. Suggestion: bad id")
+    authenticate("auth-oauth-google") {
+        get("/login") {
+
+        }
+        get("/callback") {
+            try {
+//
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt:26]: ?

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt [hunk 3]
```diff
@@ -12,27 +13,31 @@ import office.effective.features.user.service.IUserService
...
+                val principal: OAuthAccessTokenResponse.OAuth2? = call.principal()
+                call.respondText(verifier.isCorrectToken(principal!!.extraParameters["id_token"] ?: ""))
+
+            } catch (ex: Exception) {
+
+                var trace: String = ex.message ?: "There are no message.\n";
+                ex.stackTrace.forEach { trace += it.toString() + "\n" }
+                trace += "\n"
+                trace += ex.cause
+                call.respond(trace)
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt:36]: Looks weird...

Comment (1MPULSEONE) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt:36]: Put it in a separate function and use string literals

File: effectiveOfficeBackend/src/main/resources/application.conf [hunk 4]
```diff
@@ -5,11 +5,16 @@ ktor {
...
 liquibase {
-    changelogFile = "changelog/changelog-master.yaml"
-    defaultSchemaName = "public"
+  changelogFile = "changelog/changelog-master.yaml"
+  defaultSchemaName = "public"
+}
+auth {
+  user {
+    emailDomain = "gmail.com"
+  }
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/resources/application.conf:19]: Move to .gitignore immediatly, all passwords in .env

File: effectiveOfficeBackend/src/main/kotlin/office/effective/Application.kt [hunk 5]
```diff
@@ -1,19 +1,25 @@
 package office.effective
 
+//import io.ktor.server.plugins.contentnegotiation.*
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/Application.kt:3]: ?

**Review by 1MPULSEONE — CHANGES_REQUESTED** [review]

**Comment by 1MPULSEONE** [comment]
Please watch carefully for tabs and the number of blank lines in the code. Their excess or lack makes it visually "dirtier".

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt [hunk 6]
```diff
@@ -12,27 +13,31 @@ import office.effective.features.user.service.IUserService
...
-        call.respond(user ?: "No such user. Suggestion: bad id")
+    authenticate("auth-oauth-google") {
+        get("/login") {
+
+        }
+        get("/callback") {
+            try {
+//
+                val principal: OAuthAccessTokenResponse.OAuth2? = call.principal()
+                call.respondText(verifier.isCorrectToken(principal!!.extraParameters["id_token"] ?: ""))
```
Comment (1MPULSEONE) [inline effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt:28]: Validating empty string?

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/TokenVerifier.kt (hunk 1 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/TokenVerifier.kt]: Solved

File: effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt (hunk 2 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/features/user/routing/AuthRouting.kt:26]: Deleted

File: effectiveOfficeBackend/src/main/kotlin/office/effective/Application.kt (hunk 5 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/kotlin/office/effective/Application.kt]: Deleted

File: effectiveOfficeBackend/src/main/resources/application.conf (hunk 4 above)
Comment (kiselev-danil) [reply effectiveOfficeBackend/src/main/resources/application.conf]: Done

**Review by 1MPULSEONE — APPROVED** [review]

**Pull request merged**